client/firebase/authUtils.js

```diff
@@ -1,12 +1,16 @@
 
 import store, {addLocalUserInfo} from '../redux'
+import {
+  tellReduxImDoneLoading,
+  tellReduxImLoading,
+} from '../redux/reduxUtils';
 const {dispatch} = store;
 import firebase from 'firebase';
 const auth = firebase.auth();
 
   //Event listener for state changes on auth
 export function initAuth () {
-  Promise.resolve(auth.onAuthStateChanged(user => {
+  return Promise.resolve(auth.onAuthStateChanged(user => {
```

Comment:
I don't have anything against making this change, but it seems that there is no direct benefit of using 'return' since these are all side effects. Is there anything that we gain from returning the promise chain?

Reply:
I was trying to find a way to trigger the loading spinner on user sign in and sign out, as well as on app load when firebase is connection and trying to determine if a user is logged in. Initially I wanted to try chaining .then() off of the initAuth function wherever you called it. That didn't work the way I wanted, and I ended up with the current solution.

So no, nothing gained here. However, I find a generally good habit to always return your promises.

On the other hand, since auth.onAuthStateChanged is not an async function, there's no benefit gained from using Promise.resolve here at all.

Reply:
Appreciate the detailed response on this Derrick, and helping me to make sure that I wasn't missing any key conceptual thing on promise chaining. While I agree with the general principle of returning a promise, I don't think that there's any harm in strategically decided not to return a promise where a return value is not needed, but, having it in there is not an issue for me, I just wanted to make sure that I understood the purpose. 

Regarding the Promise.resolve(), we are actually getting a benefit from this. This was introduced to fix the problem of preventing our app from moving on before the sequence of redux update functions that are associated with the dispatch is complete. While, it's possible that subsequent changes to the app (such as those associated with the loading spinner) remediate this from a different angle, as far as I know, if you took out, we would see issues again where user displayName detail is getting lost for purposes of local state, due to this not getting populated on the redux store before redirect occurs.